.github/workflows/rust.yml

```diff
@@ -1,35 +1,42 @@
 name: Rust
 on:
+  # triggers when a PR is posted
   pull_request:
     branches: [develop]
+    paths-ignore:
+      - '**.md'
+  # triggers when a PR is merged
+  push:
+    branches: [develop]
+    paths-ignore:
+      - '**.md'
 jobs:
-  build-default:
+  build:
+    strategy:
+      matrix:
+        include:
+          - features: default
+          - features: rusoto-rustls
+            additional_flags: --no-default-features
+          - features: rusoto-native-tls
+            additional_flags: --no-default-features
+    env:
+       CARGO_HOME: .cargo
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    - run: rustup update stable
-    - run: cargo build --locked
-    - run: cargo test --locked
+    - uses: actions/cache@v2
+      with:
+        path: |
+          .cargo
+          target
+        # you can edit the .github/cache_bust file if you need to clear the cache
+        key: ${{ hashFiles('.github/cache_bust') }}-${{ hashFiles('.github/workflows/rust.yml') }}-${{ runner.os }}-${{ matrix.features }}-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
I think we'd want to include additional_flags in the key and restore-keys, right?  There could be a change to flags that affects builds.

Reply:
I'm not sure it would make much difference. I think in that scenario the hashes would be different and the cached bits with the wrong hashes would be ignored. My understanding of what is being accomplished here is we have 3 separate caches (one for each type of build) to prevent clobbering/cross-contamination. If we were to add additional_flags into the mix we'd just take up additional storage space by storing 3+_n_ caches.